### ISSUE TITLE
Support null-conditional operator when the return value is a pointer type

### ISSUE THREAD

**Issue by redknightlois** [dotnet/roslyn#7502]
``` csharp
public class A
{
    public byte* Ptr;
}
```

The following code will not compile.

``` csharp
var x = new A ();
byte* ptr = x?.Ptr;
```

The error itself doesnt make a lot of sense, so my guess it could be a potential bug:

`Error CS0023  -- Operator '?' cannot be applied to operand of type 'byte*'`

If not it is a quite handy operator to deal with pointers.

**Comment by gafter** [comment]
Yes, the error message should probably be more clear that it is the `?.` "operator" that cannot be applied to an operand of type 'byte*'.

**Comment by VSadov** [comment]
If this worked, it would need to return "byte*?" or Nullable<byte*>. I vaguely remember that pointers cannot be used as generic type args, so it is a CLR limitation.
UIntPtr would have worked though.
The error is indeed somewhat confusing. It should not complain about operand, the problem is with result type here.

**Comment by 333fred** [comment]
Discussed in LDM [here](https://github.com/dotnet/csharplang/blob/main/meetings/2025/LDM-2025-12-17.md#null-conditionals-evaluating-to-a-pointer-type). We will treat this as a compiler bug and fix appropriately.

**Comment by CyrusNajmabadi** [comment]
Fixed with https://github.com/dotnet/roslyn/pull/80694
